kernel/src/process.rs

```diff
@@ -1619,10 +1627,45 @@ impl<C: 'static + Chip> Process<'_, C> {
         // Minimum memory size for the process.
         let min_total_memory_size = min_app_ram_size + initial_kernel_memory_size;
 
+        // Check if this process requires a fixed memory start address. If so,
+        // try to adjust the memory region to work for this process.
+        //
+        // Right now, we only support skipping some RAM and leaving a chunk
+        // unused so that the memory region starts where the process needs it
+        // to.
+        let (new_memory_start, new_memory_size) =
```

Comment:
```suggestion
        let (requested_memory_start, requested_memory_size) =
```

"new" is confusing.., whether fixed or not, I think maybe "requested" better expresses that here it's what the loader is requesting.

Now that there are variations of "memory_start", the variables below should probably also change to be something more specific, maybe:
```diff
-let (memory_start, memory_size) = match chip.mpu().allocate_app_memory_region(
+let (actual_memory_start, actual_memory_size) = match chip.mpu().allocate_app_memory_region(
```